src/adapter/src/client.rs

```diff
@@ -536,16 +545,18 @@ impl SessionClient {
                     results.push(SimpleResult::err(
                         "executing statements of this type is unsupported via this API",
                     ));
+                    break;
                 }
             };
         }
 
-        let action = if matches!(results.last(), Some(SimpleResult::Err { .. })) {
-            EndTransactionAction::Rollback
-        } else {
-            EndTransactionAction::Commit
-        };
-        let _ = self.end_transaction(action).await?;
+        // If not in an uncommited explicit transaction, commit.
+        if !matches!(
+            self.session().transaction(),
```

Comment:
Is there a reason not to use `if self.session().transaction().is_implicit()` here? It matches protocol.rs.

Reply:
Ah, this was interacting poorly with allowing `COMMIT`/`ROLLBACK` to execute in a failed transaction. Will be fixed on next push.